experiments/ClimaEarth/run_amip.jl

```diff
@@ -588,11 +557,11 @@ cs = Interfacer.CoupledSimulation{FT}(
     (; land = land_area_fraction, ocean = zeros(boundary_space), ice = zeros(boundary_space)),
     model_sims,
     mode_specifics,
-    diagnostics,
     callbacks,
     dir_paths,
     turbulent_fluxes,
     thermo_params,
+    amip_diags_handler,
```

Comment:
we need to include `amip_diags_handler` in the `cs` object so we have access to it inside the coupling loop without creating a closure